src/django_functest/funcwebtest.py

```diff
@@ -50,13 +52,30 @@ def current_url(self):
         """
         return self.last_response.request.url
 
-    def follow_link(self, css_selector):
+    def follow_link(self, css_selector=None, text=None):
         """
-        Follows the link specified in the CSS selector.
+        Follows the link specified by CSS in css_selector= or matching the text in text=
         """
-        elems = self._make_pq(self.last_response).find(css_selector)
-        if len(elems) == 0:
-            raise WebTestNoSuchElementException(f"Can't find element matching '{css_selector}'")
+        if css_selector is not None:
+            elems = self._make_pq(self.last_response).find(css_selector)
+            if len(elems) == 0:
+                raise WebTestNoSuchElementException(f"Can't find element matching '{css_selector}'")
+        elif text is not None:
+            # cssselect (via PyQuery) handles the implementation of :contains()
+            # and it does not support nested quotation marks, so text must not
+            # contain `"`. Prefer " for the inner string so that we can still
+            # support, for instance, "can't".
```

Comment:
My experiments show that escaping `"` with `\` seems to work e.g.:
```python
>>> text = '"hello'
>>> escaped_text = text.replace('"', '\\"')
>>> escaped_text
'\\"hello'
>>> pq.find(f'span:contains("{escaped_text}")')
[<span>]
```

Could we try that here, and add a test for it?

Reply:
yes